nvflare/app_common/abstract/learner_spec.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from nvflare.apis.fl_component import FLComponent
+from nvflare.apis.fl_context import FLContext
+from nvflare.apis.shareable import ReturnCode, Shareable, make_reply
+from nvflare.apis.signal import Signal
+
+
+class Learner(FLComponent):
+    def initialize(self, parts: dict, fl_ctx: FLContext):
+        """
+        Initialize the Learner object. This is called before the Learner can train or validate.
+        This is called only once.
+
+        Args:
+            parts: components to be used by the Trainer
+            fl_ctx: FLContext of the running environment
+
+        Returns:
+
+        """
+        pass
+
+    def train(self, data: Shareable, fl_ctx: FLContext, abort_signal: Signal) -> Shareable:
+        """
+        Called to perform training. Can be called many times during the lifetime of the Learner.
+
+        Args:
+            data: the training input data (e.g. model weights)
+            fl_ctx: FLContext of the running environment
+            abort_signal: signal to abort the train
+
+        Returns: train result in Shareable
+
+        """
+        return make_reply(ReturnCode.TASK_UNKNOWN)
+
+    def get_model_for_validation(self, model_name: str, fl_ctx: FLContext) -> Shareable:
+        """
+        Called to return the best trained model from the Learner.
```

Comment:
We might need to be more precise here.
What do we mean by "best"?
And if this model is for cross-site validation, it does not need to be the best trained model.